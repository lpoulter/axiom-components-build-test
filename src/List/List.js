import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import './List.css';

export default class List extends Component {
  static propTypes = {
    /** ListItems */
    children: PropTypes.node,
    /** Style of the list and items */
    style: PropTypes.oneOf(['none', 'disc', 'ordered', 'inline']),
  };

  static defaultProps = {
    style: 'disc',
  };

  render() {
    const {
      children,
      style,
      ...rest
    } = this.props;

    const tag = style === 'ordered' ? 'ol' : 'ul';
    const classes = classnames(
      'ax-list',
      `ax-list--style-${ style }`,
    );

    return (
      <Base space="x4" { ...rest } Component={ tag } className={ classes }>
        { children }
      </Base>
    );
  }
}
