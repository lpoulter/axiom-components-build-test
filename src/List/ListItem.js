import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Base from '../Base/Base';

export default class ListItem extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <Base { ...rest } Component="li" className="ax-list__item">
        { children }
      </Base>
    );
  }
}
