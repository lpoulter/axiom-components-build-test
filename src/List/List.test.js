import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';
import ListItem from './ListItem';

const getComponent = (props = {}) =>
  renderer.create(
    <List { ...props }>
      <ListItem>Lorem Ipsum</ListItem>
    </List>
  );

describe('List', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with style', () => {
    ['disc', 'inline', 'none', 'ordered'].forEach((style) => {
      it(style, () => {
        const component = getComponent({ style });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
