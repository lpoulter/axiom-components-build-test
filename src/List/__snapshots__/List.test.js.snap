// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders with defaultProps 1`] = `
<ul
  className="ax-list ax-list--style-disc ax-space--x4"
>
  <li
    className="ax-list__item"
  >
    Lorem Ipsum
  </li>
</ul>
`;

exports[`List renders with style disc 1`] = `
<ul
  className="ax-list ax-list--style-disc ax-space--x4"
>
  <li
    className="ax-list__item"
  >
    Lorem Ipsum
  </li>
</ul>
`;

exports[`List renders with style inline 1`] = `
<ul
  className="ax-list ax-list--style-inline ax-space--x4"
>
  <li
    className="ax-list__item"
  >
    Lorem Ipsum
  </li>
</ul>
`;

exports[`List renders with style none 1`] = `
<ul
  className="ax-list ax-list--style-none ax-space--x4"
>
  <li
    className="ax-list__item"
  >
    Lorem Ipsum
  </li>
</ul>
`;

exports[`List renders with style ordered 1`] = `
<ol
  className="ax-list ax-list--style-ordered ax-space--x4"
>
  <li
    className="ax-list__item"
  >
    Lorem Ipsum
  </li>
</ol>
`;
